lib/teleterm/gatewaytest/helpers.go

```diff
@@ -128,3 +134,50 @@ func (m *MockListener) Addr() net.Addr {
 func (m *MockListener) RealAddr() net.Addr {
 	return m.realListener.Addr()
 }
+
+type KeyPairPaths struct {
+	CertPath string
+	KeyPath  string
+}
+
+func MustGenAndSaveCert(t *testing.T, opts ...alpnproxytest.SignOptionsFunc) KeyPairPaths {
+	t.Helper()
+
+	dir := t.TempDir()
+
+	ca := alpnproxytest.MustGenSelfSignedCert(t)
+
+	tlsCert := alpnproxytest.MustGenCertSignedWithCA(t, ca, opts...)
+
+	privateKey, ok := tlsCert.PrivateKey.(*rsa.PrivateKey)
+	if !ok {
+		t.Fatal("Failed to cast tlsCert.PrivateKey")
+	}
```

Comment:
```suggestion
                assert.True(t, ok, "Failed to cast tlsCert.PrivateKey")
```

Reply:
Is `assert` better than `require` here? I'd rather have the test fail if this check fails.

Reply:
If the code is not executed in a separate goroutine the `require.True` fits better here because it calls t.FailNow.

Reply:
Ah, I forgot about not calling `t.FailNow` outside of the main goroutine, thanks. So far this function has been called only from within the main goroutine and I don't think it'll change.